### PR TITLE
support external scan in parallel

### PR DESCRIPTION
**Pull request by jianwan0214**
## What type of PR is this?

- [ ] API-change
- [x] BUG
- [ ] Improvement
- [ ] Documentation
- [ ] Feature
- [ ] Test and CI
- [ ] Code Refactoring

## Which issue(s) this PR fixes:

issue #5087

## What this PR does / why we need it:
support external scan in parallel

### CODE REVIEW

File: pkg/sql/colexec/external/external.go [hunk 1]
```diff
@@ -617,15 +625,36 @@ func GetSimdcsvReader(param *ExternalParam) (*ParseLineHandler, error) {
...
 	var bat *batch.Batch
+	plh, err := getParseLineHandler(param)
+	if err != nil {
+		param.mu.Unlock()
+		return nil, err
+	}
+
+	bat, err = GetBatchData(param, plh, proc)
+	if err != nil {
+		return nil, err
```
Comment (fengttt) [inline pkg/sql/colexec/external/external.go:638]: defer unlock?   Here you are busted, I think.

File: pkg/sql/colexec/external/types.go [hunk 2]
```diff
@@ -37,11 +38,13 @@ type ExternalParam struct {
 	IgnoreLineTag int
 	// tag indicate the fileScan is finished
 	End       bool
+	Prepare   bool
```
Comment (fengttt) [inline pkg/sql/colexec/external/types.go:41]: What is this prepare?   This need a comment at least and I think, it is a really bad name -- but let me see the comment first.

**Review by fengttt — COMMENTED** [review]
The title of the PR, has nothing to do with the issue #5087.

There need to be a reasonable commmit message, telling people how do you parallelize external scan.    Parallel within one file, or many files.

File: pkg/sql/colexec/external/external.go [hunk 3]
```diff
@@ -49,11 +49,17 @@ func String(arg any, buf *bytes.Buffer) {
 
 func Prepare(proc *process.Process, arg any) error {
 	param := arg.(*Argument).Es
+	param.mu.Lock()
```
[the remark this answers is not in the document]
Comment (jianwan0214) [reply pkg/sql/colexec/external/external.go:52]: ok

**Comment by jianwan0214** [comment]
> The title of the PR, has nothing to do with the issue #5087.
> 
> There need to be a reasonable commmit message, telling people how do you parallelize external scan. Parallel within one file, or many files.

I will propose a new issue about external in parallel

**Comment by jianwan0214** [comment]
> The title of the PR, has nothing to do with the issue #5087.
> 
> There need to be a reasonable commmit message, telling people how do you parallelize external scan. Parallel within one file, or many files.

ok,I' ll do that

**Pull request closed without merging**